demux/demux_lavf.c

```diff
@@ -1122,11 +1124,13 @@ static int demux_open_lavf(demuxer_t *demuxer, enum demux_check check)
         int index = demuxer_add_chapter(demuxer, t ? t->value : "",
                                         c->start * av_q2d(c->time_base), i);
         mp_tags_copy_from_av_dictionary(demuxer->chapters[index].metadata, c->metadata);
+        av_dict_free(&c->metadata);
     }
 
     add_new_streams(demuxer);
 
     mp_tags_copy_from_av_dictionary(demuxer->metadata, avfc->metadata);
+    av_dict_free(&avfc->metadata);
```

Comment:
as this pattern is common, how about adding a parameter to `mp_tags_copy_from_av_dictionary`?